### PR TITLE
[RTM] Enh/derivatives

### LINKED ISSUE #68: Naming scheme of the outputs should follow BIDS Derivatives
See here for details: https://docs.google.com/document/d/1Wwc4A6Mow4ZPPszDIWfCUCRNstn7d_zzaWPcfcHmgI4/edit

EDIT (@oesteban): The _final product_ of this issue comprehends a couple of new tests. One test will apply to the ds005-type workflow and the other to the ds054-type workflow. The test will hard-code the desired outputs and their names, and after running the one-subject smoke tests in CircleCI, we will check that all the prescribed outputs are there with the name fulfilling BIDS-Derivatives.

For instance, the prescribed outputs for ds005 are:

```
ds005-out/
   derivatives/
       sub-01/
            anat/
                sub-01_T1w_brainmask.nii.gz
                sub-01_T1w_space-MNI152lin_brainmask.nii.gz
                sub-01_T1w_inu.nii.gz
                sub-01_T1w_dtissue.nii.gz
                sub-01_T1w_space-MNI152lin.nii.gz
                sub-01_T1w_target-MNI152lin_affine.mat
                sub-01_T1w_space-MNI152lin_class-CSF_probtissue.nii.gz
                sub-01_T1w_space-MNI152lin_class-GM_probtissue.nii.gz
                sub-01_T1w_space-MNI152lin_class-WM_probtissue.nii.gz
            func/
                sub-01_task-mixedgamblestask_run-01_bold_confounds.tsv
                sub-01_task-mixedgamblestask_run-01_bold_target-T1w_affine.txt
                sub-01_task-mixedgamblestask_run-01_bold_brainmask.nii.gz
                sub-01_task-mixedgamblestask_run-01_bold_space-MNI152lin.nii.gz
                sub-01_task-mixedgamblestask_run-01_bold_space-MNI152lin_brainmask.nii.gz
                sub-01_task-mixedgamblestask_run-01_bold_space-T1w.nii.gz
                sub-01_task-mixedgamblestask_run-01_bold_T1w-target-meanBOLD_affine.txt
                sub-01_task-mixedgamblestask_run-01_bold_hmc.nii.gz
                sub-01_task-mixedgamblestask_run-02_bold_confounds.tsv
                sub-01_task-mixedgamblestask_run-02_bold_target-T1w_affine.txt
                sub-01_task-mixedgamblestask_run-02_bold_brainmask.nii.gz
                sub-01_task-mixedgamblestask_run-02_bold_space-MNI152lin.nii.gz
                sub-01_task-mixedgamblestask_run-02_bold_space-MNI152lin_brainmask.nii.gz
                sub-01_task-mixedgamblestask_run-02_bold_space-T1w.nii.gz
                sub-01_task-mixedgamblestask_run-02_bold_T1w-target-meanBOLD_affine.txt
                sub-01_task-mixedgamblestask_run-02_bold_hmc.nii.gz
                sub-01_task-mixedgamblestask_run-03_bold_confounds.tsv
                sub-01_task-mixedgamblestask_run-03_bold_target-T1w_affine.txt
                sub-01_task-mixedgamblestask_run-03_bold_brainmask.nii.gz
                sub-01_task-mixedgamblestask_run-03_bold_space-MNI152lin.nii.gz
                sub-01_task-mixedgamblestask_run-03_bold_space-MNI152lin_brainmask.nii.gz
                sub-01_task-mixedgamblestask_run-03_bold_space-T1w.nii.gz
                sub-01_task-mixedgamblestask_run-03_bold_T1w-target-meanBOLD_affine.txt
                sub-01_task-mixedgamblestask_run-03_bold_hmc.nii.gz
```

This issue includes generating the prescribed outputs for ds054 as well for the corresponding test.

### PR DESCRIPTION
**Pull request by chrisfilo**
Rename and reformat outputs to conform with BIDS Derivatives.

Closes #68.

### CODE REVIEW

**Comment by chrisfilo** [comment]
This is how it looks now: https://circleci.com/gh/chrisfilo/fmriprep/82#artifacts/containers/0

File: fmriprep/workflows/confounds.py [hunk 1]
```diff
@@ -89,38 +91,43 @@ def discover_wf(settings, name="ConfoundDiscoverer"):
...
 
-def _gather_confounds(signals=None, dvars=None, frame_displace=None, tcompcor=None, acompcor=None):
+def _gather_confounds(signals=None, dvars=None, frame_displace=None,
+                      tcompcor=None, acompcor=None, motion=None):
     ''' load confounds from the filenames, concatenate together horizontally, and re-save '''
     import pandas as pd
     import os.path as op
 
     def less_breakable(a_string):
         ''' hardens the string to different envs (i.e. case insensitive, no whitespace, '#' '''
```
Comment (shoshber) [inline fmriprep/workflows/confounds.py]: could you fix the comment here?

**Comment by shoshber** [comment]
sorry, accidentally merged and reverted

**Comment by chrisfilo** [comment]
Please fix this.

On Sat, Dec 3, 2016 at 10:16 PM, Shoshana Berleant <notifications@github.com

**Comment by shoshber** [comment]
https://github.com/poldracklab/fmriprep/pull/212

**Comment by chrisfilo** [comment]
Changes were reverted, but there the PR dissapeared:
https://github.com/poldracklab/fmriprep/pulls

On Sat, Dec 3, 2016 at 10:26 PM, Shoshana Berleant <notifications@github.com

**Comment by chrisfilo** [comment]
I also cannot open a new PR with those changes.

**Comment by shoshber** [comment]
I reverted the reversion so it should be fine now, I think. Does this diff look right?: https://github.com/poldracklab/fmriprep/commit/c67589229e8060de61a006e95da7ad3abde28839

**Comment by chrisfilo** [comment]
It's seems good.

**Pull request merged**